src/dagre/position/bk.ts

```diff
@@ -41,12 +41,12 @@ function findType1Conflicts(g, layering) {
   var conflicts = {};
 
   function visitLayer(prevLayer, layer) {
-    var // last visited node in the previous layer that is incident on an inner
+    let // last visited node in the previous layer that is incident on an inner
       // segment.
       k0 = 0,
       // Tracks the last node in this layer scanned for crossings with a type-1
       // segment.
-      scanPos = 0,
+      scanPos = 0 as number | string,
```

Comment:
`as` is a type cast. It's safer to use a type annotation.

```suggestion
      scanPos: number | string = 0
```